CMSgov/managed-care-review
# OY2-13581: Show date when document was submitted

## Summary

Closes [13581](https://qmacbis.atlassian.net/browse/OY2-13581)

We want to show the date that a document was submitted.  Had some discussion with @macrael about the best approach here, and we decided that rather than trying to wrangle types between the old submission format and the new submission format, or coming up with a third type, we can use the information that we have in the proto buffer and calculate the date for display on the fly.  We might decide later that we want the date set on the document object.

#### Screenshots

<img width="773" alt="image" src="https://user-images.githubusercontent.com/8964335/160877511-f0046642-0535-483e-9a3c-8d9da8be2138.png">


#### Test cases covered

Added what amounts to an integration test for SubmissionSummary, and a smaller test for the UploadedDocumentsTable, which is doing the actual date display.

## QA guidance

Note that the displayed date should be the date of submission, not upload.

And as I opened a few test files, I saw linter errors in my editor, so I addressed them, which is why you see a bunch of apparently unrelated tests in this PR.
